modules/publication/ajax/FileUpload.php

```diff
@@ -0,0 +1,934 @@
+                            'PublicationCollaboratorID' => $cid,
+                           );
+        $db->insertIgnore(
+            'publication_collaborator_rel',
+            $collabRelInsert
+        );
+    }
+}
+
+/**
+ * Inserts users with edit access for the given publication project
+ *
+ * @param int $pubID publication ID
+ *
+ * @return null
+ */
+function insertEditors($pubID)
+{
+    if (empty($_REQUEST['usersWithEditPerm'])) {
+        return;
+    }
+
+    $db = Database::singleton();
+    $usersWithEditPerm = json_decode($_REQUEST['usersWithEditPerm']);
+    foreach ($usersWithEditPerm as $uid) {
+        $insert = array(
+                   'PublicationID' => $pubID,
+                   'UserID'        => $uid,
+                  );
+
+        $db->insertIgnore(
+            'publication_users_edit_perm_rel',
+            $insert
+        );
+    }
+}
+
+/**
+ * Inserts keywords for the given publication project
+ *
+ * @param int $pubID publication ID
+ *
+ * @return null
+ */
+function insertKeywords($pubID)
+{
+    if (empty($_REQUEST['keywords'])) {
+        return;
+    }
+    $db       = Database::singleton();
+    $keywords = json_decode($_REQUEST['keywords']);
+    foreach ($keywords as $kw) {
+        // check if keyword exists
+        $kwID = $db->pselectOne(
+            'SELECT PublicationKeywordID ' .
+            'FROM publication_keyword ' .
+            'WHERE Label=:kw',
+            array('kw' => $kw)
+        );
+        // if it doesn't, add it to keyword table and retrieve ID
+        if (!$kwID) {
+            $kwInsert = array('Label' => $kw);
+            $db->insert(
+                'publication_keyword',
+                $kwInsert
+            );
+            $kwID = $db->pselectOne(
+                'SELECT PublicationKeywordID ' .
+                'FROM publication_keyword ' .
+                'WHERE Label=:kw',
+                array('kw' => $kw)
+            );
+        }
+        // add it pub_kw_rel table
+        // get publication ID
+        $pubKWRelInsert = array(
+                           'PublicationID'        => $pubID,
+                           'PublicationKeywordID' => $kwID,
+                          );
+
+        $db->insert(
+            'publication_keyword_rel',
+            $pubKWRelInsert
+        );
+
+    }
+}
+
+/**
+ * Inserts Variables of Interest for the given publication project
+ *
+ * @param int $pubID publication ID
+ *
+ * @return null
+ */
+function insertVOIs($pubID)
+{
+    if (empty($_REQUEST['voiFields'])) {
+        return;
+    }
+    $db        = Database::singleton();
+    $testNames = $db->pselectColWithIndexKey(
+        'SELECT ID, Test_name FROM test_names',
+        array(),
+        'ID'
+    );
+
+    $paramTypes = $db->pselectColWithIndexKey(
+        'SELECT ParameterTypeID, Name FROM parameter_type',
+        array(),
+        'ParameterTypeID'
+    );
+
+    $voiFields = json_decode($_REQUEST['voiFields']);
```

Comment:
As discussed, variables of interest from the imaging side should be linkable as well.